src/odemis/driver/xt_client.py

```diff
@@ -2215,3 +2237,90 @@ def _setMultiBeamMode(self, multi_beam_mode):
         #     self.apertureIndex.value = current_aperture
         #     self.beamletIndex.value = current_beamlet
         return (self.parent.get_use_case() == 'MultiBeamTile')
+
+
+class MultiBeamDetector(Detector):
+    """
+    This is an extension of xt_client.Detector class. It overwrites the image acquisition
+    to work with image acquisition in XTToolkit.
+    """
+
+    def __init__(self, name, role, parent, channel_name, address, **kwargs):
+        """
+        channel_name (str): Name of one of the electron channels.
+        """
+        # The acquisition is based on a FSM that roughly looks like this:
+        # Event\State |    Stopped    |   Acquiring    | Receiving data |
+        #    START    | Ready for acq |        .       |       .        |
+        #    DATA     |       .       | Receiving data |       .        |
+        #    STOP     |       .       |     Stopped    |    Stopped     |
+        #    TERM     |     Final     |      Final     |     Final      |
+        self._proxy_access = threading.Lock()
```

Comment:
This lock is to protect "cancel_connection", so name it more similarly. Like `_cancel_access` or `_cancel_connection_lock`...